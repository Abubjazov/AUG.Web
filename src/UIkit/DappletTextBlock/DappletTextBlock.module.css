.root {
  display: flex;
  flex-direction: column;
  animation: visible 0.5s ease-in-out;
}

.title {
  font-size: 14px;
  font-weight: 400;
  line-height: 14px;
  color: var(--graylight);
  text-align: left;
}

.text {
  margin-top: 5px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--textgray);
  text-align: left;
}
