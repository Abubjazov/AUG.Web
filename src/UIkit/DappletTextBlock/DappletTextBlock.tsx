import { FC } from 'react'

import { combineClasses as cc } from 'utils/combineClasses'

import styles from './DappletTextBlock.module.css'

export interface DappletTextBlockProps {
  userStyles?: string
  title: string
  text: string
}

const DappletTextBlock: FC<DappletTextBlockProps> = ({
  userStyles = '',
  title,
  text,
}) => {
  return (
    <div className={cc([styles.root, userStyles])}>
      <span className={styles['title']}>{title}</span>

      <span className={styles['text']}>{text}</span>
    </div>
  )
}

export default DappletTextBlock
