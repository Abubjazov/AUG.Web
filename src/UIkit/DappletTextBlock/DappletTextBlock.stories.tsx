import type { Meta, StoryObj } from '@storybook/react'

import DappletTextBlock, { DappletTextBlockProps } from './DappletTextBlock'

import '/src/index.css'

const meta: Meta<DappletTextBlockProps> = {
  component: DappletTextBlock,
  decorators: [
    Story => (
      <div style={{ width: '400px' }}>
        <Story />
      </div>
    ),
  ],
  title: 'UIkit/Dapplet Text Block',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - basic dapplet text group',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    title: { description: 'Text group title' },
    text: { description: 'Text group text' },
  },
}

export default meta

type Story = StoryObj<DappletTextBlockProps>

export const Default: Story = {
  args: {
    title: 'Title',
    text: 'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
  },
}
