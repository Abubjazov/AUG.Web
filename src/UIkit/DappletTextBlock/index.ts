import DappletTextBlock from './DappletTextBlock'

export default DappletTextBlock
