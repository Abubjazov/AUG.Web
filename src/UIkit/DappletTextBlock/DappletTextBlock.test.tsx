import { render } from '@testing-library/react'

import DappletTextBlock from './DappletTextBlock'

describe('DappletTextBlock', () => {
  test('should render DappletTextBlock default', () => {
    const { asFragment } = render(
      <DappletTextBlock title={'Title'} text={'Some text'} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
