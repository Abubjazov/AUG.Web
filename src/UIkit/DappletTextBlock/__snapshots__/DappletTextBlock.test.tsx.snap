// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DappletTextBlock > should render DappletTextBlock default 1`] = `
<DocumentFragment>
  <div
    class="_root_94add5 "
  >
    <span
      class="_title_94add5"
    >
      Title
    </span>
    <span
      class="_text_94add5"
    >
      Some text
    </span>
  </div>
</DocumentFragment>
`;
